### PR TITLE
Fix Windows Action Mode tab label truncation

### PR DESCRIPTION
**Pull request by myleshorton**
Windows could show “Action Mo…” in the desktop tab bar even though unused space remained around the pill. The fixed per-tab inset reduced the label's available width before Flutter applied ellipsis.

Measure the localized label using the current font and text scale, and shrink the desktop inset when needed to accommodate the icon, full label, and status dot. Keep the preferred inset wherever it fits and retain ellipsis when the tab genuinely lacks room.

Validation: all 295 Flutter tests pass; targeted Dart analysis and `git diff --check` pass. New Windows-platform widget tests check actual text truncation at 1× and 2× text scale; both fail against the original implementation and pass with this fix. Native Windows execution was not available on the macOS development host.

Report: https://wdynhnkxvsdx.slack.com/archives/C06F5EKHN5S/p1789127321860339

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved desktop navigation spacing so labels and controls fit more reliably across different window widths and text sizes.
  - Improved mobile navigation layout by removing unnecessary spacing around navigation items.

- **Tests**
  - Expanded coverage for desktop navigation across multiple viewport widths and text scales.
  - Added validation for label truncation behavior in narrow layouts.
  - Added checks for rendering errors and correct navigation sizing.

### CODE REVIEW

File: test/features/share_my_connection/action_mode_navigation_test.dart [hunk 1]
```diff
@@ -38,6 +39,48 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  for (final (width, scale) in [(460.0, 1.0), (780.0, 2.0)]) {
+    testWidgets('desktop label uses available inset space at scale $scale', (
```
[the remark this answers is not in the document]
Comment (myleshorton) [reply test/features/share_my_connection/action_mode_navigation_test.dart]: Verified by running the tests against origin/main: both the 460px/1× and 780px/2× assertions fail before the fix and pass after it. These widget tests use Ahem, whose 11-character Action Mode label measures 154px at 14px and 308px at 2×. The original layout leaves only 118px and 298px respectively, so both exercise inset shrinking. Commit 74c171230 documents that font geometry and adds the narrow boundary case. All 296 tests pass.

File: test/features/share_my_connection/action_mode_navigation_test.dart [hunk 2]
```diff
@@ -38,6 +39,48 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  for (final (width, scale) in [(460.0, 1.0), (780.0, 2.0)]) {
```
[the remark this answers is not in the document]
Comment (myleshorton) [reply test/features/share_my_connection/action_mode_navigation_test.dart]: Added 393px/1× coverage in 74c171230. The expected result at that width is ellipsis: these tests use Ahem, so the full label plus fixed content needs 210px, exceeding the 180.5px available even with zero inset. This now verifies the genuine-space-shortage fallback; the 460px/1× and 780px/2× cases verify complete rendering after inset reduction. Both of those cases were rerun against the original implementation and fail there. All 296 tests pass.
